CDTDatastoreTests/CDTSessionCookieInterceptorTests.m

```diff
@@ -324,4 +356,146 @@ - (void)testCookieInterceptorRenewsEarly
+            return [OHHTTPStubsResponse
+                    responseWithJSONObject:@{
+                                             @"ok" : @(YES),
+                                             @"name" : @"username",
+                                             @"roles" : @[ @"_admin" ]
+                                             }
+                    statusCode:200
+                    headers:@{
+                              @"Set-Cookie" : [NSString
+                                               stringWithFormat:@"%@; Version=1; Path=/; HttpOnly; Max-Age=86400",
+                                               testCookieHeaderValue2]
+                              }];
+        }];
+        
+        // get resource successfully using new cookie
+        [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+            XCTAssert([testCookieHeaderValue2 isEqualToString: request.allHTTPHeaderFields[@"Cookie"]]);
+            XCTAssert([request.HTTPMethod isEqualToString:@"GET"]);
+            return [OHHTTPStubsResponse responseWithJSONObject:@{@"ok" : @(YES)} statusCode:200 headers:@{}];
+        }];
+    } else {
+        // get resource fails, cookie was not re-newed and is empty
+        [helper addResponse:^OHHTTPStubsResponse *__nonnull(NSURLRequest *__nonnull request) {
+            XCTAssertNil(request.allHTTPHeaderFields[@"Cookie"]);
+            XCTAssert([request.HTTPMethod isEqualToString:@"GET"]);
+            return [OHHTTPStubsResponse responseWithJSONObject:@{@"error": error,
+                                                                 @"reason": reason}
+                                                    statusCode:403 headers:@{}];
+        }];
+    }
+
+    
+    [helper doStubsForHost:@"username403.cloudant.com"];
+    
+    CDTSessionCookieInterceptor *interceptor =
+    [[CDTSessionCookieInterceptor alloc] initWithUsername:@"username403" password:@"password"];
+    // create a context with a request which we can use
+    NSURL *url = [NSURL URLWithString:@"http://username403.cloudant.com/somedb"];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url];
+    
+    CDTURLSession *session = [[CDTURLSession alloc] initWithCallbackThread:[NSThread currentThread] requestInterceptors:@[interceptor] sessionConfigDelegate: nil];
+    
+    // Initial GET will get a cookie and resource
+    CDTURLSessionTask *task = [session dataTaskWithRequest:request taskDelegate:nil];
+    [task resume];
+    while ([task state] != NSURLSessionTaskStateCompleted) {
+        [NSThread sleepForTimeInterval:1.0];
+    }
+    
+    // Second GET should fail with 403 error
+    CDTURLSessionTask *task2 = [session dataTaskWithRequest:request taskDelegate:nil];
+    [task2 resume];
+    while ([task2 state] != NSURLSessionTaskStateCompleted) {
+        [NSThread sleepForTimeInterval:1.0];
+    }
+    
+    if(interceptor.shouldMakeSessionRequest == NO) {
```

Comment:
As per above, if it means what I think it does as opposed to what you think it does then this check won't work once you fix it, you'll need to use something else to distinguish the test paths.

Reply:
Fixed in 9b2f6bb.